main.go

```diff
@@ -35,18 +35,27 @@ func main() {
 
 	// init logger
 	logger := log.NewJSONLogger(os.Stdout)
-	logger = log.With(
-		logger,
+	logger = log.With(logger,
 		"ts", log.DefaultTimestampUTC,
+		"caller", log.DefaultCaller,
 	)
-	logger = leveled.NewFromEnv(logger)
-	logger = log.With(logger, "caller", log.DefaultCaller)
+	// set the allowed log level filter
+	switch strings.ToLower(os.Getenv("LOGLEVEL")) {
+	case "debug":
+		level.NewFilter(logger, level.AllowDebug())
```

Comment:
Small mistake: You need to assing the return value of level.NewFilter to the `logger` variable or else it has no effect.